src/ngraph/runtime/dynamic/dynamic_backend.cpp

```diff
@@ -120,130 +120,206 @@ bool runtime::dynamic::DynamicExecutable::call(
     // (1) all shapes;
     // (2) all values of shape-relevant input tensors.
 
-    NGRAPH_CHECK(m_wrapped_function->get_parameters().size() == inputs.size());
-
-    std::vector<std::shared_ptr<runtime::Tensor>> wrapped_inputs;
-    std::vector<element::Type> arg_element_types;
-    std::vector<PartialShape> arg_shapes;
-
-    std::shared_ptr<Function> clone;
+    std::vector<int> merged_input_shapes;
+    std::ostringstream key;
+    size_t loop_count = 0;
+    for (auto& input : inputs)
     {
-        // We'll use AlignedBuffers to back the base pointers, storing them in this vector for RAII
-        // purposes.
-        std::vector<AlignedBuffer> arg_buffers;
-        arg_buffers.reserve(inputs.size());
-        std::vector<void*> arg_value_base_pointers(inputs.size());
-
-        size_t i = 0;
-
-        for (auto& input : inputs)
+        if (m_wrapped_function->get_parameters()[loop_count]->is_relevant_to_shapes())
         {
-            if (m_wrapped_function->get_parameters()[i]->is_relevant_to_shapes())
+            // Caching on values of Shape relevant inputs
+            void* data_ptr = malloc(input->get_size_in_bytes());
+            input->read(data_ptr, input->get_size_in_bytes());
+            // assuming that shape relevant inputs are int64_t
+            int64_t* result = (int64_t*)data_ptr;
+            for (int i = 0; i < input->get_element_count(); i++)
```

Comment:
Just allocate an int64_t* directly; you can pass that to anything that needs a void* and it will be properly aligned. No need for this casting. Use a unique_ptr for it and then we don't need to worry about freeing it in the future if someone rearranges the control flow.